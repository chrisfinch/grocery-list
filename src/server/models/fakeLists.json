{
  "lists": [
    {
      "id": "1",
      "title": "Demo list",
      "created": 1564695374960,
      "items": [
        {
          "name": "beans",
          "purchased": false
        },
        {
          "name": "bread",
          "purchased": false
        },
        {
          "name": "broccoli",
          "purchased": false
        },
        {
          "name": "cereal",
          "purchased": false
        },
        {
          "name": "squash",
          "purchased": false
        }
      ]
    },
    {
      "id": "2",
      "title": "Other list",
      "created": 1564695541598,
      "items": [
        {
          "name": "pasta",
          "purchased": false
        }
      ]
    }
  ]
}