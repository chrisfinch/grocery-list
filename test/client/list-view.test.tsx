import React from 'react';
import { shallow } from 'enzyme';
import ListView from '../../src/client/list-view';
import { lists as fakeLists } from '../fakeLists.json';

describe('List View', () => {
    beforeEach(() => {
        fetchMock.resetMocks();
    });

    it('Renders Correctly', () => {
        fetchMock.mockResponseOnce(JSON.stringify(fakeLists[0]));

        const ListWrapper = shallow(<ListView activeListId={null} />);

        expect(ListWrapper).toMatchSnapshot();

        ListWrapper.setProps({ activeListId: 0 });
        ListWrapper.update();
        ListWrapper.instance().forceUpdate();

        expect(ListWrapper).toMatchSnapshot();

        expect(fetchMock.mock.calls.length).toEqual(1);
        expect(fetchMock.mock.calls[0][0]).toEqual('/lists/0');
    });
});
