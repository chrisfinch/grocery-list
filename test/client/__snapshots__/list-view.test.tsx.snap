// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List View Renders Correctly 1`] = `""`;

exports[`List View Renders Correctly 2`] = `""`;
